### ISSUE TITLE
Allow for multiple download locations for different operating systems

### ISSUE THREAD

**Issue by moritzheiber** [mislav/bump-homebrew-formula-action#5]
Linuxbrew is a thing:tm: thus some Homebrew formula are doing stuff like this:

```ruby
# The main formula
class MyFormula < Formula
  # [...]
  version '0.0.1'
  # [...]

  if OS.mac?
    url 'https://some.url/v0.0.1/binary-macos'
    sha256 'ce06501c7f791b4d653effb50f234cd6fa451c734054acb9936c14e965abf37d'
  elsif OS.linux?
    url 'https://some.url/v0.0.1/binary-linux'
    sha256 'a64cc55300ccd8491759fa5a7df1031e57b83d9712512333f792d4fb448dd766'
  end
  # [...]
end

```

It would be cool to be able to support this formula with this action :slightly_smiling_face:

**Comment by mislav** [comment]
@moritzheiber Yes that would be a great feature! It's just that, without a proper ruby parser, it would be really hard to parse out the `if..else` clauses to figure out where to inject the different URLs/checksums 🤔

**Comment by rami3l** [comment]
@mislav I want this feature too, but... Why should we need a parser in this case?

In my opinion, maybe we can use stuff like magic comments:

```ruby
class Foo < Formula
  [ .. ]
  # BEGIN VERSION
  version "v0.1.0-alpha.1"
  # END VERSION
  [ .. ]
end
```

... and I think we might be able to write a script which generates `ruby` code and uses it to fill in the blanks.

**Comment by rami3l** [comment]
@moritzheiber @mi Oh no, I just thought too much: why can't we add support for **template files**?

**Comment by moritzheiber** [comment]
@rami3l Here be dragon! :dragon: :fire:

**Comment by karwan19** [comment]
class Foo < Formula
  [ .. ]
  # BEGIN VERSION
  version "v0.1.0-alpha.1"
  # END VERSION
  [ .. ]
end

**Comment by karwan19** [comment]
# BEGIN URL.MAC
url "https://some.url/v0.0.1/binary-macos"
# END URL.MAC

**Comment by karwan19** [comment]
LdeCVtfy72ze1Ch7UKGS

![Image](https://github.com/user-attachments/assets/83adaf31-c12f-4b82-a88a-9720d9bb9d8b)
